src/main/java/org/primefaces/util/AgentUtils.java

```diff
@@ -30,30 +30,35 @@
     private AgentUtils() {
     }
 
-    public static boolean isEdge(FacesContext context) {
-        String userAgent = context.getExternalContext().getRequestHeaderMap().get("User-Agent");
+    private static String getUserAgent(FacesContext context) {
+        return context.getExternalContext().getRequestHeaderMap().get("User-Agent");
+    }
+
+    private static boolean userAgentContains(FacesContext context, String fragment) {
+        String userAgent = getUserAgent(context);
 
         if (userAgent == null) {
             return false;
         }
         else {
-            return userAgent.contains("Edge");
+            return userAgent.contains(fragment);
         }
     }
 
-    public static boolean isIE(FacesContext context) {
-        String userAgent = context.getExternalContext().getRequestHeaderMap().get("User-Agent");
+    public static boolean isEdge(FacesContext context) {
+        return userAgentContains(context, "Edge");
+    }
 
-        if (userAgent == null) {
-            return false;
-        }
-        else {
-            return userAgent.contains("MSIE");
-        }
+    public static boolean isIE(FacesContext context) {
+        return userAgentContains(context, "MSIE");
     }
 
+    /**
+     * @deprecated it is unused
+     */
+    @Deprecated(since = "11.0.0")
```

Comment:
Or should I simply delete these methods? Or will that be part of removing IE support?

Reply:
We will make them part of the other ticket. I will add a note there.